theano/tensor/nnet/nnet.py

```diff
@@ -802,6 +803,9 @@ def softmax_graph(c):
 
 
 def softmax(c):
+    c = as_tensor_variable(c)
+    if c.broadcastable[-1]:
+        warnings.warn("The softmax is appllied on a dimension of shape 1, which doesn't have a semantic meaning.")
```

Comment:
"applied" with one "l"
"does not" is better than "doesn't" in error messages in general